lib/enum/arrays.rb

```diff
@@ -1,24 +1,31 @@
 class Array
 
   def sum_of_numeric_elements
+    self.select { |num| num.is_a? (Numeric) }.reduce(:+)
   end
 
   def product_of_numeric_elements
+    self.select { |num| num.is_a? (Numeric) }.reduce(:*)
   end
 
   def even_numeric_elements
+    self.select { |num| num.is_a? Integer }.select { |num| num.even? }
```

Comment:
Better to use Numeric here instead of Integer - otherwise you would exclude floats like 2.0